go/host/enclave/guardian.go

```diff
@@ -608,3 +646,40 @@ func (g *Guardian) streamEnclaveData() {
 		}
 	}
 }
+
+func (g *Guardian) getBatchesAfterSize(seqNo uint64) (uint64, error) {
+	var size uint64
+
+	currentNo := seqNo
+	for {
+		batch, err := g.sl.L2Repo().FetchBatchBySeqNo(big.NewInt(int64(currentNo)))
+		if err != nil {
+			if errors.Is(err, errutil.ErrNotFound) {
+				break // no more batches
+			}
+			return 0, err
+		}
+
+		bSize, err := batch.Size()
+		if err != nil {
+			return 0, err
+		}
+		size += uint64(bSize)
```

Comment:
I don't know enough about the rollup sizing to know if what this is doing is a good proxy for what the limiter does on the enclave side (question for @tudor-malene perhaps).

Does adding up these batch sizes give a reasonable sense of how big the rollup would be? Even accounting for compression if you had a long string of empty batches for example (like does Size() give zero for empty batch?).

Guess it doesn't need to be perfect, if it's overestimating the size then we'll be rolling up slightly more often than necessary sometimes, if it's underestimating then it'll catch up. We'll know more once we've got some live stats I guess.

Just want to make sure there aren't edge cases that could make this method dramatically misbehave.

Reply:
I think it's fine. We can always adjust the thresholds to more conservative values
We'll see in practice.